language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -0,0 +1,493 @@
+  private final Map<URI, String> languageIdPerFileURI = new HashMap<>();
+  private final SonarLintTelemetry telemetry = new SonarLintTelemetry();
+
+  private Path workspaceDir;
+  private String testFilePattern;
+  private Map<String, String> analyzerProperties;
+  private int ruleServerPort;
+
+  public SonarLintLanguageServer(int port) throws IOException {
+    Socket socket = new Socket("localhost", port);
+
+    Launcher<LanguageClient> launcher = LSPLauncher.createServerLauncher(this,
+      socket.getInputStream(),
+      socket.getOutputStream(),
+      true, new PrintWriter(System.out));
+
+    this.client = launcher.getRemoteProxy();
+    this.logOutput = new LanguageClientLogOutput(client);
+
+    info("Starting SonarLint engine...");
+
+    try {
+      Builder builder = StandaloneGlobalConfiguration.builder()
+        .setLogOutput(logOutput)
+        .addPlugins(getAnalyzers().toArray(new URL[0]));
+
+      this.engine = new StandaloneSonarLintEngineImpl(builder.build());
+    } catch (Exception e) {
+      error("Error starting SonarLint engine", e);
+      throw new IllegalStateException(e);
+    }
+
+    info("SonarLint engine started");
+
+    backgroundProcess = launcher.startListening();
+  }
+
+  private void debug(String message) {
+    client.logMessage(new MessageParams(MessageType.Log, message));
+  }
+
+  private void info(String message) {
+    client.logMessage(new MessageParams(MessageType.Info, message));
+  }
+
+  private void warn(String message) {
+    client.logMessage(new MessageParams(MessageType.Warning, message));
+  }
+
+  public void error(String message, Throwable t) {
+    StringWriter sw = new StringWriter();
+    PrintWriter pw = new PrintWriter(sw);
+    t.printStackTrace(pw);
+    client.logMessage(new MessageParams(MessageType.Error, message + "\n" + sw.toString()));
+  }
+
+  private Collection<URL> getAnalyzers() throws IOException, URISyntaxException {
+    info("Load analyzers...");
+    List<URL> plugins = new ArrayList<>();
+    URI uri = SonarLintLanguageServer.class.getResource("/plugins").toURI();
+    Path myPath;
+    if ("jar".equals(uri.getScheme())) {
+      try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+        myPath = fileSystem.getPath("/plugins");
+        collect(plugins, myPath);
+      }
+    } else {
+      myPath = Paths.get(uri);
+      collect(plugins, myPath);
+    }
+
+    if (plugins.isEmpty()) {
+      throw new IllegalStateException("Found no analyzers in server");
+    }
+    info("Found " + plugins.size() + " analyzers");
+    return plugins;
+  }
+
+  private static void collect(List<URL> plugins, Path myPath) throws MalformedURLException, IOException {
+    try (Stream<Path> walk = Files.walk(myPath, 1)) {
+      for (Iterator<Path> it = walk.iterator(); it.hasNext();) {
+        Path file = it.next();
+        if (file.toString().toLowerCase().endsWith(".jar")) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1449)